### ISSUE TITLE
eslint creating error `Parsing error: Parenthesized pattern`

### ISSUE THREAD

**Issue by suyashdb** [eslint/eslint#9705]
**Tell us about your environment**
Mac OS
* **ESLint Version:**  "eslint": "^3.16.1",
* **Node Version:** node v6.10.2

* **npm Version:** npm  v3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?**
default 
**Please show your full configuration:**
```
{
    "env": {
        "browser": true,
        "commonjs": true,
        "node": true
    },
    "globals": {
        "Uint8Array": false,
        "describe": false,
        "before": false,
        "it": false
    },
    "extends": "eslint:recommended",
    "rules": {
        "no-console": [
            "error",
            {"allow": ["warn"]}
        ],
        "linebreak-style": [
            "error",
            "unix"
        ],
        "semi": [
            "error",
            "always"
        ]
    }
}
```
<details>
<summary>Configuration</summary>

```js



```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```
var groupBy = function(array, cp){
    return array.reduce((groups, next) => { const name = cp(next);
    if (name !== undefined) {
        const group = groups.get(name);
        if (group) group.push(next);
        else groups.set(name, [next]);
    }
    return groups;
}, new Map());
}

```

```
npm run lint

```

**What did you expect to happen?**
no lint error for ` return array.reduce((groups, next) => { const name = cp(next);`
**What actually happened? Please include the actual, raw output from ESLint.**
gave following error:

```
  19:25  error  Parsing error: Parenthesized pattern

✖ 1 problem (1 error, 0 warnings)


npm ERR! Darwin 15.6.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "lint"
npm ERR! node v6.10.2
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! bids-validator@0.0.0 lint: `./node_modules/eslint/bin/eslint.js ./**/*.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the bids-validator@0.0.0 lint script './node_modules/eslint/bin/eslint.js ./**/*.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the bids-validator package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     ./node_modules/eslint/bin/eslint.js ./**/*.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs bids-validator
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls bids-validator
npm ERR! There is likely additional logging output above.

```

**Comment by platinumazure** [comment]
I don't see you using ES6 in your configuration. Without ES6, you can't use arrow functions.

Try adding this to your config: `"parserOptions": { "ecmaVersion": 6 }`

**Comment by suyashdb** [comment]
It adds more error:

```
26:8   error  'Map' is not defined  no-undef
32:21  error  'Set' is not defined  no-undef
✖ 4 problems (4 errors, 0 warnings)
```

**Comment by platinumazure** [comment]
At least it's parsing successfully now. :smile:

In order to recognize the es6 globals, simply add `"es6": true` to your environments config.

**Comment by suyashdb** [comment]
`"es6": true ` doesn't help.. it solves the parsing error but other two are still coming up..

**Comment by platinumazure** [comment]
Can you please share exactly what errors you're seeing now? (And please
share the full program if you haven't already done so, thanks!)

**Comment by suyashdb** [comment]
```
  26:8   error  'Map' is not defined  no-undef
  32:21  error  'Set' is not defined  no-undef

✖ 2 problems (2 errors, 0 warnings)


npm ERR! Darwin 15.6.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "lint"
npm ERR! node v6.10.2
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! bids-validator@0.0.0 lint: `./node_modules/eslint/bin/eslint.js ./**/*.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the bids-validator@0.0.0 lint script './node_modules/eslint/bin/eslint.js ./**/*.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the bids-validator package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     ./node_modules/eslint/bin/eslint.js ./**/*.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs bids-validator
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls bids-validator
npm ERR! There is likely additional logging output above.

```
git repo link: https://github.com/suyashdb/bids-validator/tree/NoBIDS_subspecificFiles

**Comment by platinumazure** [comment]
Try running `./node_modules/eslint/bin/eslint.js --print-config path/to/file.js` and see if es6: true shows up in the environments.

Just to be clear, this is how the es6 environment should be added:

```json
{
    "env": {
        "browser": true,
        "node": true,
        "commonjs": true,
        "es6": true
    }
}
```

**Comment by suyashdb** [comment]
actually it doesnt show up:
with above command i am getting following:

```
"env": {
    "browser": true,
    "commonjs": true,
    "node": true
  },
```

**Comment by platinumazure** [comment]
Please try running ESLint with the `--debug` flag to see if it will tell you what configuration file(s) it is loading. Maybe the config file you are editing is not being loaded, or maybe a different config file is being loaded with higher precedence/priority than the one you are working with.

**Comment by not-an-aardvark** [comment]
Closing because there hasn't been a response in awhile -- feel free to create a new issue if you're still having trouble.
